kratos/processes/apply_ray_casting_interface_recognition_process.h

```diff
@@ -0,0 +1,191 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Daniel Diez
+//
+
+#pragma once
+
+// System includes
+#include <string>
+#include <iostream>
+
+// External includes
+
+// Project includes
+#include "processes/apply_ray_casting_process.h"
+
+namespace Kratos
+{
+///@addtogroup Kratos Core
+///@{
+
+///@name Kratos Classes
+///@{
+
+/// Applies ray casting to distinguish the color (like in/out) of each node in modelpart
+/** This class is used to define the which nodes are inside, outside or on the interface
+ *  of certain volume described by its contour
+ */
+template<std::size_t TDim = 3>
+class KRATOS_API(KRATOS_CORE) ApplyRayCastingInterfaceRecognitionProcess : public ApplyRayCastingProcess<TDim>
+{
+
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of ApplyRayCastingInterfaceRecognitionProcess
+    KRATOS_CLASS_POINTER_DEFINITION(ApplyRayCastingInterfaceRecognitionProcess);
+
```

Comment:
Considering that now we have #10911, I'd take the chance to register it. See `define_registry.h`, but long story short we need to add
```c++
    KRATOS_REGISTRY_ADD_PROTOTYPE("Processes.KratosMultiphysics", ApplyRayCastingInterfaceRecognitionProcess<TDim>)
    KRATOS_REGISTRY_ADD_PROTOTYPE("Processes.All", ApplyRayCastingInterfaceRecognitionProcess<TDim>)
```

Reply:
done. It seems we need to define default ctor for this and the base class so registry works...

Reply:
Yes. A default ctor is required in order to create the prototype.